packages/autocomplete-js/src/utils/focusAndOpenKeyboard.ts

```diff
@@ -0,0 +1,30 @@
+import { AutocompleteEnvironment } from '@algolia/autocomplete-core';
+
+// Mobile devices don't open the keyboard when programmatically focusing inputs.
+// See https://stackoverflow.com/a/55425845
+export function focusAndOpenKeyboard(
+  environment: AutocompleteEnvironment,
+  inputElement: HTMLInputElement,
+  timeout: number = 0
+) {
+  if (inputElement) {
+    const tempInput = environment.document.createElement('input');
+    const { top, left } = inputElement.getBoundingClientRect();
+
+    tempInput.style.position = 'absolute';
```

Comment:
personally I'd rather see this as intentional browser behaviour instead of trying hacky fixes. I like the refactor and simplifications in the main file, but personally wouldn't do the temporary input

Reply:
I copied exactly the hack, I'll check if this works without.

Reply:
I meant that I'm not fan of having a hack in a library, I feel like it might have unexpected side effects in the future

Reply:
Yeah, and it currently does actually—there's a regression introduced by the fact that we no longer batch, and it might require additional code which will increase the library's size.

I suggest we close this and mark as wontfix for now. We can revisit later if it resurfaces.